src/commands/deployAzurePipeline/configureCicdPipeline/models/aksCluster.ts

```diff
@@ -0,0 +1,13 @@
+import { GenericResource } from 'azure-arm-resource/lib/resource/models';
+import { CloudExplorerV1 } from 'vscode-kubernetes-tools-api';
+
+export class aksCluster implements GenericResource {

```

Comment:
❓ Wondering how extending `GenericResource` cater what is discussed here? : https://github.com/Azure/vscode-aks-tools/pull/42#discussion_r499921565

Reply:
I think we should _extend_ GenericResource and not _implement_ it

Reply:
We cannot extend the GenericResource interface.
![image](https://user-images.githubusercontent.com/58769601/95331101-1e9c3580-08c7-11eb-8f39-e4bb390ba47c.png)

Reply:
Actually extend might not even be needed here. You can compose new interface by including GenericResource and subscriptionId.